### PR TITLE
chore(QTM-302): Updates react to 17 version

### PR DESCRIPTION
**Pull request by toncabral88**
## Description
https://jirasoftware.catho.com.br/browse/QTM-302

- Removes enzyme from project;
- Updates react and react-dom;
- Updates @babel/preset-react and jest;
- Removes react imports from components, stories and tests;

## Setup
to view the components behavior, run `yarn storybook`

## Review guide
- [ ] Coverage (coverage status should not regress)\
      - `yarn test:coverage`
- [ ] Unit tests (yarn test:components)
- [ ] Regression \
      - first start the storybook for regression tests(and keep it open): ` yarn test:regression:storybook`; \
      - then run the regression tests: `yarn test:regression`
- [ ] Code review

### Browsers review
- [ ] Layout review
  - [ ] Edge
  - [ ] Firefox
  - [ ] Chrome
  - [ ] Safari
  - [ ] Mobile

### CODE REVIEW

**Review by tcelestino — APPROVED** [review]
- [x] Coverage (coverage status should not regress)\
      - `yarn test:coverage`
- [x] Unit tests (yarn test:components)
- [x] Regression \
      - first start the storybook for regression tests(and keep it open): ` yarn test:regression:storybook`; \
      - then run the regression tests: `yarn test:regression`
- [x] Code review

### Browsers review
- [ ] Layout review
  - [x] Edge
  - [x] Firefox
  - [ ] Chrome
  - [ ] Safari
  - [ ] Mobile

**Review by matheusrpp — APPROVED** [review]
## Review guide
- [x] Coverage (coverage status should not regress)\
      - `yarn test:coverage`
- [x] Unit tests (yarn test:components)
- [x] Regression \
      - first start the storybook for regression tests(and keep it open): ` yarn test:regression:storybook`; \
      - then run the regression tests: `yarn test:regression`
- [x] Code review

File: components/List/sub-components/Content.jsx [hunk 1]
```diff
@@ -26,13 +26,13 @@ class Content extends React.Component {
     }
 
     return (
-      <React.Fragment>
+      <>
```
Comment (rapahaeru) [inline components/List/sub-components/Content.jsx:29]: =]

File: package.json [hunk 2]
```diff
@@ -91,9 +91,6 @@
     "cross-env": "^5.2.0",
     "cz-conventional-changelog": "2.1.0",
     "dotenv-webpack": "^1.6.0",
-    "enzyme": "^3.11.0",
```
Comment (rapahaeru) [inline package.json:94]: bye

**Review by rapahaeru — APPROVED** [review]
## Review guide
- [x] Coverage (coverage status should not regress)\
      - `yarn test:coverage`
- [x] Unit tests (yarn test:components)
- [ ] Regression \
      - first start the storybook for regression tests(and keep it open): ` yarn test:regression:storybook`; \
      - then run the regression tests: `yarn test:regression`
- [x] Code review

### Browsers review
- [x] Layout review
  - [ ] Edge
  - [ ] Firefox
  - [ ] Chrome
  - [ ] Safari
  - [ ] Mobile

**Pull request merged**